### ISSUE TITLE
Use Qbs everywhere and remove the qmake project files

### ISSUE THREAD

**Issue by bjorn** [mapeditor/tiled#1484]
Since it is annoying to keep both the Qbs and the qmake projects up to date with new files, we should move forward with only one system. Qbs is already the only supported build method on Windows, however the following tasks are open before we can remove the qmake projects:

- [x] Build macOS packages using Qbs
- [x] Build Linux binary packages using Qbs
- [x] Update translation files `.ts` with Qbs or possibly using some script (currently, `make ts` is run in the `translations` folder)
- [x] Make sure Linux distributions can use Qbs for building their package
- [x] Build the tests using Qbs (currently only included in qmake project).
- [x] Build the snap using Qbs

This issue is here to document parts still depending on qmake and what the Qbs project files are possibly still lacking, which we'll find out as we try to move from qmake to Qbs in the above areas.

**Comment by bjorn** [comment]
@Ablu If you have some time, you could help with the 4th item on the list by trying to change the Fedora package of Tiled to be built with Qbs. :-)

**Comment by bjorn** [comment]
@svenstaro Maybe you'd like to help make sure Qbs is ready (or find out in what way it is not) by making the Arch Linux package build with Qbs. I'll try it myself eventually in any case, so feel free to wait as well.

**Comment by jakepetroules** [comment]
FYI: The Arch Linux package recently moved from AUR to an official package in Extras.

**Comment by svenstaro** [comment]
Can you provide some packaging-friendly qbs instructions to build tiled?

**Comment by bjorn** [comment]
@svenstaro It seems to essentially work by replacing the `build()` and `package()` functions in your package as follows:
```sh
build() {
  cd $srcdir/$pkgname-$pkgver
  qbs build --no-install
}

package() {
  cd $pkgname-$pkgver
  qbs qbs.installRoot:$pkgdir/usr
}
```
And of course adding `qbs` to the `makedepends`.

However, looking at the file list it is clear that I've got some work to do, since the desktop file, icons, man pages, mime stuff and thumbnailer all still need to be added. Also I'll probably need to add an option to install the header files, and possibly also to disable RPATH (is that needed?).

@jakepetroules One difference I noted is that qmake will link to /usr/lib/libtiled.so.1.0.0 and symlink /usr/lib/libtiled.so, /usr/lib/libtiled.so.1 and /usr/lib/libtiled.so.1.0, while Qbs just produces a /usr/lib/libtiled.so file. Is the qmake behavior necessary in this context, and does Qbs have an option for this?

**Comment by jakepetroules** [comment]
@bjorn Set the version property of your dynamic library product and you should get the symlinks.

**Comment by bjorn** [comment]
I've pushed some changes that should cover most things related to building the packages, like installing all the same files that the qmake project does, and including options to install the header files and disable the use of rpath.

**Comment by bjorn** [comment]
> @bjorn Set the version property of your dynamic library product and you should get the symlinks.

Ok, thanks! Are they necessary though? I don't provide any form of binary compatibility for libtiled.

**Comment by jakepetroules** [comment]
Nope, Qbs just tries to follow platform conventions by default.

**Comment by bjorn** [comment]
@svenstaro So, updated instructions would be:
```sh
build() {
  cd $srcdir/$pkgname-$pkgver
  qbs --no-install Tiled.installHeaders:true Tiled.useRPaths:false
}

package() {
  cd $pkgname-$pkgver
  qbs qbs.installRoot:$pkgdir/usr Tiled.installHeaders:true Tiled.useRPaths:false
}
```
I'm wondering, how important is it to separate the build and package states? Cause it could just be done in a single command (only the second one is needed).

> Nope, Qbs just tries to follow platform conventions by default.

For now I guess I won't set a version then, since it has no meaning at all in terms of binary compatibility.

**Comment by Ablu** [comment]
> @Ablu If you have some time, you could help with the 4th item on the list by trying to change the Fedora package of Tiled to be built with Qbs. :-)

sure. But I fear this needs a bit of work. I would guess that Tiled is the first project in Fedora which builds with qbs only then. But it is a good thing. So I will try to reach out on the maillinglist to get some input for it after I am done with my exams.

**Comment by Ablu** [comment]
> I'm wondering, how important is it to separate the build and package states? Cause it could just be done in a single command (only the second one is needed).

hm... both RPM and DEB based builds usually first build into a intermediate directory... The install step then only moves files from the intermediate build result to the final destination... But I will ask about this too...

**Comment by bjorn** [comment]
@Ablu @TingPing Now that the qmake project files are gone, the stuff in [`dist/copr`](https://github.com/mapeditor/tiled/tree/master/dist/copr) should be either updated or removed. Please let me know what would be more appropriate.

**Comment by Ablu** [comment]
I fear I currently do not have the resources to getting packaging things fixed with Qbs (especially since Qbs is now superseded by cmake). So I would just vote for removing those unless there is someone else to maintain this.
